### PR TITLE
SALTO-2515: SF deployment of editable properties in StandardFields

### PR DESCRIPTION
**Pull request by tamtamirr**
Fixed a bug when deploying changes to editable properties of standard fields.

---

This PR turned to be way more simple than suspected. Been playing around with the artifacts we deploy, and turns out that omitting the field `type` from `CustomField` upon deploy allows deploying successfully.

---
_Release Notes_: 
Salesforce Adapter:
- Implemented the ability to deploy editable properties of StandardFields.
---
_User Notifications_: 
_None_

### CODE REVIEW

File: packages/salesforce-adapter/src/client/types.ts [hunk 1]
```diff
@@ -103,7 +103,7 @@ export interface LookupFilter {
 export class CustomField implements MetadataInfo {
   // Common field annotations
   readonly label?: string
-  readonly type: string
+  readonly type?: string
```
Comment (tamtamirr) [inline packages/salesforce-adapter/src/client/types.ts:106]: @ori-moisis Should we just omit this field completely? Does it make any sense to leave it like that?

Comment (ori-moisis) [reply packages/salesforce-adapter/src/client/types.ts:106]: what do you mean?
how will we create new fields with the correct type if we never send the type?

Comment (tamtamirr) [reply packages/salesforce-adapter/src/client/types.ts:106]: @ori-moisis As for deploying fields under `CustomObject` it's not required. (Not that it's not required, it caused the bug 🙈), So I guess we could leave it as optional

**Comment by tamtamirr** [comment]
Will reopen this PR with a more simple solution.

**Pull request closed without merging**